### PR TITLE
coverage while upgrading linter

### LINKED ISSUE #5974: Help upgrading the linter
PR #5966 tries to upgrade the linter, but runs into many new lint issues that need to be fixed before upgrade can be merged. Most of them are about signed/unsigned conversions, which we generally force-ignore.

Success criteria: `gh pr checkout 5966` and `make lint`

### PR DESCRIPTION
**Pull request by anishbista60**
## Which problem is this PR solving?
fixes #5974 

## Description of the changes
- fixing the problem created during upgrade of the linter

## How was this change tested?
- 

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -143,6 +143,7 @@ linters-settings:
       - G107
       - G404
       - G601
+      - G115
```
Comment (yurishkuro) [inline .golangci.yml:146]: we don't want to turn the rule completely

Comment (anishbista60) [reply .golangci.yml:146]: @yurishkuro Then, should i use 
```
//nolint: gosec // G115
```
Instead?

Comment (yurishkuro) [reply .golangci.yml:146]: yes

**Comment by anishbista60** [comment]
@yurishkuro Done. Works fine in my machine.

File: cmd/agent/app/reporter/grpc/flags.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ func (b *ConnBuilder) InitFromViper(v *viper.Viper) (*ConnBuilder, error) {
 	if hostPorts != "" {
 		b.CollectorHostPorts = strings.Split(hostPorts, ",")
 	}
+	//nolint: gosec // G115
 	b.MaxRetry = uint(v.GetInt(retryFlag))
```
Comment (yurishkuro) [inline cmd/agent/app/reporter/grpc/flags.go:43]:
```suggestion
	b.MaxRetry = v.GetUint(retryFlag)
```

File: cmd/collector/app/span_handler_builder.go [hunk 3]
```diff
@@ -51,6 +51,7 @@ func (b *SpanHandlerBuilder) BuildSpanProcessor(additional ...ProcessSpan) proce
 		Options.NumWorkers(b.CollectorOpts.NumWorkers),
 		Options.QueueSize(b.CollectorOpts.QueueSize),
 		Options.CollectorTags(b.CollectorOpts.CollectorTags),
+		//nolint: gosec // G115
 		Options.DynQueueSizeWarmup(uint(b.CollectorOpts.QueueSize)), // same as queue size for now
```
Comment (yurishkuro) [inline cmd/collector/app/span_handler_builder.go:55]: we should change CollectorOpts.QueueSize to uint instead (negative queue size is invalid)

File: plugin/storage/es/options.go [hunk 4]
```diff
@@ -330,6 +330,7 @@ func initFromViper(cfg *namespaceConfig, v *viper.Viper) {
 	cfg.UseReadWriteAliases = v.GetBool(cfg.namespace + suffixReadAlias)
 	cfg.Enabled = v.GetBool(cfg.namespace + suffixEnabled)
 	cfg.CreateIndexTemplates = v.GetBool(cfg.namespace + suffixCreateIndexTemplate)
+	//nolint: gosec // G115
 	cfg.Version = uint(v.GetInt(cfg.namespace + suffixVersion))
```
Comment (yurishkuro) [inline plugin/storage/es/options.go:334]:
```suggestion
	cfg.Version = v.GetUint(cfg.namespace + suffixVersion)
```

**Comment by anishbista60** [comment]
@yurishkuro done

**Review by yurishkuro — APPROVED** [review]
Thanks!

**Pull request merged**